btcjson/chainsvrresults.go

```diff
@@ -632,3 +639,50 @@ type EstimateSmartFeeResult struct {
 	Errors  []string `json:"errors,omitempty"`
 	Blocks  int64    `json:"blocks"`
 }
+
+var _ json.Unmarshaler = &FundRawTransactionResult{}
+
+type rawFundRawTransactionResult struct {
+	Transaction    string  `json:"hex"`
+	Fee            float64 `json:"fee"`
+	ChangePosition int     `json:"changepos"`
+}
+
+// FundRawTransactionResult is the result of the fundrawtransaction JSON-RPC call
+type FundRawTransactionResult struct {
+	Transaction    *wire.MsgTx
+	Fee            btcutil.Amount
+	ChangePosition int // the position of the added change output, or -1
+}
```

Comment:
I think there's a pretty clear benefit of decoding the hex for the user. Proper types makes it easier to do to the right thing, and the user gets access to a bunch of useful methods and fields. If they want the raw bytes or hex, they can call `Transaction.Serialize` on the returned struct. 

In your eyes, what's the drawback of deserializing the hex?

Reply:
I'm not against deserializing the hex, I'm suggesting we can simplify implementation and keep the code consistent by adding a new struct and replacing `*wire.MsgTx`, I was thinking something like:
```go
type DeserializedTransaction struct {
    Version  int32  `json:"version"`
    Vin      []Vin  `json:"vin"`
    Vout     []Vout `json:"vout"`
    LockTime uint32 `json:"locktime"`
}
```
This would have the benefit of 1. using the already-written `Vin` and `Vout` json serialization methods, and 2 (more important imo) it's consistent with the rest of the chain server result structs. It's just a suggestion, but I personally think it would be a positive change :)

Reply:
Sorry, I'm not sure I understand what you mean. 

This is the output you get when funding raw TXs with Bitcoin Core: 

```bash
$ bitcoin-cli -regtest fundrawtransaction 02000000000100e1f5050000000017a914fbb59111ea0588622c8b752bd3af4f044c83075e8700000000
{
  "hex": "02000000015a79c7d84a745dc2fda83fe21c29977396e765b9728502297bb6c0eaea5aa3d20000000000feffffff0200e1f5050000000017a914fbb59111ea0588622c8b752bd3af4f044c83075e8798415a030000000017a914e8e26462a80694e11c02b8ba9fd81da0ffaf55548700000000",
  "fee": 0.00003320,
  "changepos": 1
}
```

It seems to me like parsing the `hex` field into a `wire.MsgTx` is the easiest solution, both for the implementors and users. `wire.MsgTx` has the fields you listed above: 

```go
type MsgTx struct {
	Version  int32
	TxIn     []*TxIn
	TxOut    []*TxOut
	LockTime uint32
}
```

Reply:
Sorry, it was my bad - I forgot that `fundrawtransaction` has no verbose flag unlike `getrawtransaction`, my idea was to take advantage of the server sending the tx info when verbose is set but there is no verbose flag here so it won't work. LGTM